files/build_templates/swss_vars.j2

```diff
@@ -2,5 +2,6 @@
     "asic_type": "{{ asic_type }}",
     "asic_id": "{{ DEVICE_METADATA.localhost.asic_id }}",
     "mac": "{{ DEVICE_METADATA.localhost.mac }}",
-    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %}
+    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %},
+    "resource_type": "{{ DEVICE_METADATA.localhost.resource_type }}"
```

Comment:
do we know if resource_type is always there or not, will this throw exception if resource_type is not defined?

Reply:
In my understanding, it will come as `"resource_type": ""` if not defined. @nazariig can you confirm.

Reply:
@lguohan / @prsunny please have a look at this example:
* **no value:**
```bash
root@sonic:/home/admin# cat /etc/sonic/config_db.json | jq -r '.DEVICE_METADATA'
{
  "localhost": {
    "buffer_model": "traditional",
    "default_bgp_status": "up",
    "default_pfcwd_status": "disable",
    "hostname": "sonic",
    "hwsku": "ACS-MSN4600",
    "mac": "1c:34:da:1e:95:00",
    "platform": "x86_64-mlnx_msn4600-r0"
  }
}

root@sonic:/# cat /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "{{ asic_type }}",
    "asic_id": "{{ DEVICE_METADATA.localhost.asic_id }}",
    "mac": "{{ DEVICE_METADATA.localhost.mac }}",
    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %},
    "resource_type": "{{ DEVICE_METADATA.localhost.resource_type }}"
}

root@sonic:/# sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "mellanox",
    "asic_id": "",
    "mac": "1c:34:da:1e:95:00",
    "synchronous_mode": "enable",
    "resource_type": ""
}
```

* **default value:**
```bash
root@sonic:/home/admin# redis-cli -n 4 HSET 'DEVICE_METADATA|localhost' resource_type default
(integer) 1

root@sonic:/# sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "mellanox",
    "asic_id": "",
    "mac": "1c:34:da:1e:95:00",
    "synchronous_mode": "enable",
    "resource_type": "default"
}
```